enumerables.rb

```diff
@@ -0,0 +1,197 @@
+      my_each do |arg|
+        next if yield arg
+
+        return false
+      end
+    elsif !expr
+      my_each do |arg|
+        next if arg
+
+        return false
+      end
+    elsif expr.is_a? Regexp
+      my_each do |arg|
+        next if arg.match(expr)
+
+        return false
+      end
+    elsif expr.class == Class
+      my_each do |arg|
+        next if arg.is_a?(expr)
+
+        return false
+      end
+    else
+      my_each do |arg|
+        next if arg == expr
+
+        return false
+      end
+    end
+    true
+  end
+
+  def my_any?(expr = nil) # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+    if block_given?
+      my_each do |arg|
+        next unless yield arg
+
+        return true
+      end
+    elsif !expr
+      my_each do |arg|
+        next unless arg
+
+        return true
+      end
+    elsif expr.is_a? Regexp
+      my_each do |arg|
+        next if arg.match(expr).nil?
+
+        return true
+      end
+    elsif expr.class == Class
+      my_each do |arg|
+        next unless arg.is_a?(expr)
+
+        return true
+      end
+    else
+      my_each do |arg|
+        next unless arg == expr
+
+        return true
+      end
+    end
+    false
+  end
+
+  def my_none?(expr = nil) # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+    if block_given?
+      my_each do |arg|
+        next unless yield arg
+
+        return false
+      end
+    elsif !expr
+      my_each do |arg|
+        next unless arg
+
+        return false
+      end
+    elsif expr.is_a? Regexp
+      my_each do |arg|
+        next if arg.match(expr).nil?
+
+        return false
+      end
+    elsif expr.class == Class
+      my_each do |arg|
+        next unless arg.is_a?(expr)
+
+        return false
+      end
+    else
+      my_each do |arg|
+        next unless arg == expr
+
+        return false
+      end
+    end
+    true
+  end
+
+  def my_count(arg = nil)
+    counter = 0
+    if arg
+      my_each { |param| counter += 1 if param == arg }
+    elsif block_given?
+      my_each { |param| counter += 1 if yield(param) }
+    else
+      counter = size
+    end
+    counter
+  end
+
+  def my_map(arg = nil)
+    return to_enum unless block_given?
+
+    arr = []
+    if arg
+      my_each_with_index { |elem, val| arr[val] = arr.call(elem) }
+    else
+      my_each_with_index { |elem, val| arr[val] = yield(elem) }
+    end
+    arr
+  end
+
+  def my_inject(initial = nil, sym = nil, &block) # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+    sum = initial || first
+    sum = first if initial.class == Symbol
+    sum = initial if !initial.class == Symbol
+    sum -= sum if sum != initial && sum.class != String
+    my_each { |e| sum = block.call(sum, e) } if block_given?
+    my_each { |e| sum = sum.send(sym, e) } if initial && sym
+    my_each { |e| sum = sum.send(initial, e) } if initial.class == Symbol
+    sum
+  end
```

Comment:
**[REQUIRED]**
* Combines all elements of enum by applying a binary operation, specified by a block.
```
Failure/Error: expect(actual).to eq expected
     
       expected: 5069015533618896340602101361010794807396273594826752000000000000
            got: 4
     
       (compared using ==)
```
* When a symbol is specified combines each element of the collection by applying the symbol as a named method.
```
Failure/Error: expect(range.my_inject(2, :*)).to eq range.inject(2, :*)
     
       expected: 2534507766809448170301050680505397403698136797413376000000000000
            got: 2
     
       (compared using ==)
```